### PR TITLE
Upgrade Apache Maven to 3.9.6

### PR DESCRIPTION
**Pull request by dmikurube**

### CODE REVIEW

**Review by hiroyuki-sato — APPROVED** [review]
LGTM👍(One minor comment)

File: gradle.lockfile [hunk 1]
```diff
@@ -17,24 +17,24 @@ org.apache.commons:commons-compress:1.10=embulk-decoder-bzip2,embulk-encoder-bzi
...
+org.apache.maven:maven-artifact:3.9.6=deps
+org.apache.maven:maven-builder-support:3.9.6=deps
+org.apache.maven:maven-model-builder:3.9.6=deps
+org.apache.maven:maven-model:3.9.6=deps
+org.apache.maven:maven-repository-metadata:3.9.6=deps
+org.apache.maven:maven-resolver-provider:3.9.6=deps
 org.codehaus.plexus:plexus-interpolation:1.26=deps
 org.codehaus.plexus:plexus-utils:3.5.1=deps
-org.eclipse.sisu:org.eclipse.sisu.inject:0.3.5=deps
+org.eclipse.sisu:org.eclipse.sisu.inject:0.9.0.M2=deps
```
[the remark this answers is not in the document]
Comment (dmikurube) [reply gradle.lockfile:37]: Yeah I am aware of that, but Apache Maven 3.9.6 (`maven-model-builder:3.9.6`) depends exactly on Sisu 0.9.0 M2, and Sisu does not have any newer versions.

* https://mvnrepository.com/artifact/org.apache.maven/maven-model-builder/3.9.6
* https://mvnrepository.com/artifact/org.eclipse.sisu/org.eclipse.sisu.inject

We have no other choice.

[the remark this answers is not in the document]
Comment (hiroyuki-sato) [reply gradle.lockfile:37]: I got it. Thanks

**Comment by dmikurube** [comment]
The `NOTICE` file should have been updated in #1643 -- Eclipse Sisu should have been NOTICEd.

**Comment by dmikurube** [comment]
Thanks!

**Pull request merged**